### PR TITLE
Offline

### PR DESCRIPTION
**Pull request by tomermoshe**

### CODE REVIEW

File: src/app/chat/services/channels/channels.service.ts [hunk 1]
```diff
@@ -16,7 +16,11 @@ export class ChannelsService {
       userId: user.id
     };
 
-    return this.apollo.watchQuery<MyChannelsQuery.Result>({query: myChannelsQuery, variables});
+    return this.apollo.watchQuery<MyChannelsQuery.Result>({
+      query: myChannelsQuery,
+      variables,
+      fetchPolicy: 'cache-and-network',
```
Comment (eitanfr) [inline src/app/chat/services/channels/channels.service.ts:22]: 1. there is a way to set a global fetch policy
2. they removed `network-first` hope it won't cause a problem

File: src/app/chat/services/chat/chat.service.ts [hunk 2]
```diff
@@ -69,10 +69,11 @@ export class ChatService {
     this.messagesQueryObservable = this.apollo.watchQuery<MessagesQuery.Result>({
       query: messagesQuery,
       variables: messagesQueryVariables,
+      fetchPolicy: 'cache-and-network',
     });
 
-    return this.messagesQueryObservable.do(({ data }) => {
-      if (data.messages) {
+    return this.messagesQueryObservable.do(({ data, loading}) => {
+      if (!loading && data.messages) {
```
Comment (eitanfr) [inline src/app/chat/services/chat/chat.service.ts:76]: does loading ever becomes true? 
checkout the doc 
http://dev.apollodata.com/angular2/queries.html#basics

File: src/app/shared/services/login-page.service.ts [hunk 3]
```diff
@@ -20,4 +22,9 @@ export class LoginPageService {
       },
     });
   }
+
+  cleanCache() {
```
Comment (eitanfr) [inline src/app/shared/services/login-page.service.ts]: lets put it somewhere else it not really  login page logic

File: src/app/shared/services/login-page.service.ts [hunk 4]
```diff
@@ -20,4 +22,9 @@ export class LoginPageService {
...
+    getPersistor().purge();
```
Comment (eitanfr) [inline src/app/shared/services/login-page.service.ts]: lets put it somewhere else . it not really belongs to login page logic

File: src/app/shared/components/login-page/login-page.component.ts [hunk 5]
```diff
@@ -25,6 +25,7 @@ export class LoginPageComponent implements OnInit {
   }
 
   ngOnInit() {
+    this.loginService.cleanCache();
```
Comment (eitanfr) [inline src/app/shared/components/login-page/login-page.component.ts]: maybe we should move it to "after the user has a successful login ". like after resume session

File: src/app/chat/services/chat/chat.service.ts (hunk 2 above)
Comment (tomermoshe) [reply src/app/chat/services/chat/chat.service.ts:76]: yes yes

File: src/app/chat/services/channels/channels.service.ts (hunk 1 above)
Comment (tomermoshe) [reply src/app/chat/services/channels/channels.service.ts:22]: penis

File: src/app/shared/components/login-page/login-page.component.ts (hunk 5 above)
Comment (tomermoshe) [reply src/app/shared/components/login-page/login-page.component.ts]: done

File: src/app/shared/services/login-page.service.ts (hunk 4 above)
Comment (tomermoshe) [reply src/app/shared/services/login-page.service.ts]: done

File: src/app/shared/services/login-page.service.ts (hunk 3 above)
Comment (tomermoshe) [reply src/app/shared/services/login-page.service.ts]: done

**Pull request merged**